src/compiler/checker.ts

```diff
@@ -18707,14 +18707,15 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
         if (accessNode) {
             const indexNode = getIndexNodeForAccessExpression(accessNode);
-            if (indexType.flags & (TypeFlags.StringLiteral | TypeFlags.NumberLiteral)) {
+            if (indexNode.kind !== SyntaxKind.BigIntLiteral && indexType.flags & (TypeFlags.StringLiteral | TypeFlags.NumberLiteral)) {
                 error(indexNode, Diagnostics.Property_0_does_not_exist_on_type_1, "" + (indexType as StringLiteralType | NumberLiteralType).value, typeToString(objectType));
             }
             else if (indexType.flags & (TypeFlags.String | TypeFlags.Number)) {
                 error(indexNode, Diagnostics.Type_0_has_no_matching_index_signature_for_type_1, typeToString(objectType), typeToString(indexType));
             }
             else {
-                error(indexNode, Diagnostics.Type_0_cannot_be_used_as_an_index_type, typeToString(indexType));
+                const typeString = typeToString(indexNode.kind === SyntaxKind.BigIntLiteral ? getBaseTypeOfLiteralType(getContextFreeTypeOfExpression(indexNode)) : indexType);

```

Comment:
We should either write the value, or `bigint` (all lowercase) as it's the underlying base type.

Reply:
Sorry, I've been looking at the tests, and some of the test cases have been inconsistent with the caps. I'll fix that.

Would you prefer the "property does not exist on type" error? I was using the "bigint literal type cannot be used as an index" error because the [original design choice](https://github.com/microsoft/TypeScript/issues/15096#issuecomment-438751487) was the bigintliteral type cannot be used as an index, as you should use a string or number instead. "cannot be used as index" seemed more descriptive of what was happening than "property does not exist". 

I don't mind using the value in the error message (`1n cannot be used as an index type`) if you like that better, and it is what you would get with `typeToString(indexType)`. I changed it because `bigint` made the above more clear as well.

Reply:
cc @rbuckton @jakebailey @DanielRosenwasser

Reply:
I guess if you just swap it back to `bigint`, that should be sufficient. I see what you mean about the inconsistency; it's not that say `1n` doesn't exist on the type, but that bigint isn't supported at all?

Reply:
Yes, it's valid syntactically in js to index with bigints, but we've disallowed them when type checking

Reply:
It's `bigint` not `bigInt`